File: cogs/utility/avatar_history_cog.py
import discord
from discord.ext import commands
from utils.database import fetch_avatar_history, fetch_name_history
from utils.error_handler import error_handler
from utils.permissions import has_permissions_to_send_images

class AvatarHistoryCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self.local_cache = bot.local_cache

    @commands.command(name='avatar')
    @error_handler
    @has_permissions_to_send_images()
    async def avatar_command(self, ctx, action: str, user: discord.User = None):
        if user is None:
            user = ctx.author

        if action.lower() == 'history':
            avatar_history = self.fetch_avatar_history(user.id)
            if avatar_history:
                await ctx.send(f"Here's the avatar history for {user.name}: {', '.join(avatar_history)}")
            else:
                await ctx.send(f"Sorry, I couldn't find any avatar history for {user.name}.")
        elif action.lower() == 'current':
            await ctx.send(f"Hey there! Here's the current avatar for {user.name}: {user.avatar_url}")
        else:
            await ctx.send("Invalid action. Please choose 'history' or 'current'.")

    def fetch_avatar_history(self, user_id):
        return fetch_avatar_history(user_id, self.local_cache)

    def fetch_name_history(self, user_id):
        return fetch_name_history(user_id, self.local_cache)

def setup(bot):
    bot.add_cog(AvatarHistoryCog(bot))

File: cogs/utility/avatar_history_cog.ts
import { Client, Message, User } from 'discord.js-selfbot-v13';
import { Command } from 'discord-akairo';
import { errorHandler } from '../../utils/error_handler';
import { hasPermissionsToSendImages } from '../../utils/permissions';
import DatabaseManager from '../../utils/database';

class AvatarHistoryCog {
    private client: Client;
    private databaseManager: DatabaseManager;

    constructor(client: Client) {
        this.client = client;
        this.databaseManager = new DatabaseManager();
    }

    @Command({
        aliases: ['avatar'],
        description: 'Fetch avatar history or current avatar of a user.',
        args: [
            { id: 'action', type: 'string' },
            { id: 'user', type: 'user', default: null }
        ]
    })
    @errorHandler
    @hasPermissionsToSendImages()
    async avatarCommand(message: Message, { action, user }: { action: string, user: User | null }) {
        if (!user) {
            user = message.author;
        }

        try {
            if (action.toLowerCase() === 'history') {
                const avatarHistory = await this.databaseManager.fetch_avatar_history(user.id);
                if (avatarHistory.length > 0) {
                    await message.channel.send(`Here's the avatar history for ${user.username}: ${avatarHistory.join(', ')}`);
                } else {
                    await message.channel.send(`Sorry, I couldn't find any avatar history for ${user.username}.`);
                }
            } else if (action.toLowerCase() === 'current') {
                await message.channel.send(`Hey there! Here's the current avatar for ${user.username}: ${user.displayAvatarURL()}`);
            } else {
                await message.channel.send("Invalid action. Please choose 'history' or 'current'.");
            }
        } catch (error) {
            console.error(`Failed to fetch avatar history. Error: ${error.message}`);
            await message.channel.send(`An error occurred while fetching avatar history for ${user.username}. Please try again later.`);
        }
    }
}

export function setup(client: Client) {
    return new AvatarHistoryCog(client);
}
